programs/svm-spoke/src/state/refund_account.rs

```diff
@@ -0,0 +1,131 @@
+use anchor_lang::prelude::*;
+use anchor_spl::token_interface::TokenAccount;
+
+use crate::error::CustomError;
+
+#[account]
+#[derive(InitSpace)]
+pub struct ClaimAccount {
+    pub amount: u64,
+}
+
+// When executing relayer refund leaf, refund accounts are passed as remaining accounts and can hold either a regular
```

Comment:
does this mean that, in theory, a relayer could close their ATA right before the bundle execution tx lands on-chain, thereby making the execution of the bundle in-valid as the remaining account type is wrong? so when the executor looks at the tx to send they build TokenAccount, expecting that the relayer has an ATA but before the tx lands on-chain the relayer can front run the tx and delete their ATA, which then requires, for this method to work, that the executor passes in the ClaimAccount, not token account, which breaks the bundle execution and the whole thing reverts?

In theory, this means that a relayer who can front run the execution can always "toggle" the status of their ATA (opening it or closing it), thereby blocking the bundle execution?

if the executor sends these execution transactions via private mempool they might be able to protect themselves (you cant close/open the account before the execution) but this is not going to be failsafe?

thoughts on this?